pages/agent/v3/configuration.md.erb

```diff
@@ -231,7 +231,12 @@ _Optional attributes:_
     <tr>
       <th><code>name</code></th>
       <td>
-        The name of the agent. Supports the template variables <code>%hostname</code> (the agent machine’s hostname), <code>%n</code> (a unique number for the agent), and <code>%pid</code> (the agent process id).
+        The name of the agent. Supports the template variables
+        <ul>
+        <li><code>%hostname</code> (the agent machine’s hostname)</li>
+        <li><code>%n</code> (a unique number for the agent)</li>
+        <li><code>%pid</code> (the agent process id)</li>
+         Note that if you're using <code>--spawn</code> to run multiple agents in a single process, they all share the same <code>%pid</code>. Make sure that each agent running on a host with the same <code>build-path</code> has a unique name.
         <p class="Docs__api-param-eg"><em>Default:</em> <code>"%hostname-%n"</code></p>
         <p class="Docs__api-param-eg"><em>Environment variable:</em> <code>BUILDKITE_AGENT_NAME</code></p>
```

Comment:
We should add the `%spawn` one, with the note that it requires the latest agent version:

```suggestion
        <li><code>%hostname</code> (the agent machine’s hostname)</li>
        <li><code>%spawn</code> (a unique number for each agent started using <code>--spawn</code>; added in <a href="https://github.com/buildkite/agent/releases/tag/v3.27.0">v3.27.0</a>)</li>
        <li><code>%n</code> (a unique number for the agent)</li>
        <li><code>%pid</code> (the agent process id)</li>
        Note that if you're using <code>--spawn</code> to run multiple agents in a single process, we recommend using <code>%spawn</code> in your agent name, or to ensure that each agent running on a host with the same <code>build-path</code> has a unique name.
        <p class="Docs__api-param-eg"><em>Default:</em> <code>"%hostname-%n"</code></p>
        <p class="Docs__api-param-eg"><em>Default on Elastic CI Stack for AWS:</em> <code>"%hostname-%spawn"</code> (<a href="https://github.com/buildkite/elastic-ci-stack-for-aws/releases/tag/v5.2.0">v5.2.0</a> and later)</p>
        <p class="Docs__api-param-eg"><em>Environment variable:</em> <code>BUILDKITE_AGENT_NAME</code></p>
```

I've also added a note about the default now being different in the Elastic Stack.